tests/async_tests/test_connection_pool.py

```diff
@@ -53,14 +57,59 @@ def is_connection_dropped(self) -> bool:
         return False
 
 
+class AlwaysPendingConnection(MockConnection):
```

Comment:
This class is required since it's possible to reuse pending http2 connection. It's difficult to emulate is without our local test server introduced by @florimondmanca in https://github.com/encode/httpcore/pull/205